torch/csrc/jit/scope.h

```diff
@@ -45,5 +47,41 @@ struct TORCH_API Scope : public c10::intrusive_ptr_target {
   std::string namesFromRoot(const std::string& separator = "/") const;
 };
 
+/**
```

Comment:
I am not sure this is tracking the right thing. First, we already know that any node inside function F has F on the callstack, so it seems redundant and self-referential to store that. An empty callstack could just indicate that the Node has not been inlined anywhere and so it adds no _extra_ nodes to the call stack when we are reporting an error.   Also, we should not _just_ be recording the function, but also the SourceRange of the invocation that got inlined. e.g.

```
def foo():
   print("foo")

def bar()
   print("hi")
   foo() # call me
   print("bye")


# after inlining 
def bar()
    print("hi") # SourceRange("print("hi")")
    print("foo") # {SourceRange("foo() # call me"), SourceRange("print("foo")"), }
    print("bar") #  SourceRange("print("bar")")
```

When we report an exception happening on a line we append all the source range information into the call stack we report and then report the rest of the trace of the interpreter stack.

Reply:
@zdevito, please take another look! There is no test yet, but I'll work on it in the meanwhile (as per our discussion, I plan to add an internal function that would return the current callstack - that's how we would expose callstacks in tests).

Reply:
I've added a test - I realized that it can be easily done if I don't try to expose callstack to python and just do it in C++. The "ShowCallstack" node that we discussed might still be useful, but now it should not be blocking this PR.